### PR TITLE
Add GitHub Actions export only shell

### LINKED ISSUE #903: [Question] Using direnv in GitHub Actions
I spent quite some time getting this to work, so wanted to share it here. Maybe (after som revisions) it can be added to the Wiki.

We wanted to use `direnv` in GitHub Actions, without using a third-party action.

Here's the result:

```yaml
  - name: Install direnv
    run: |
      mkdir -p ~/.local/bin
      curl -sfL https://direnv.net/install.sh | bash
      direnv allow

  - name: Run Tool
    run: |
      eval "$(direnv export bash)"
     * run some tool installed by direnv *
```

**Note** that we're running this in a self-hosted Ubuntu runner. YMMV

Unfortunately you have to duplicate `eval "$(direnv export bash)"` for every step.

So a question is: Is there a better way of solving that? Maybe involving `$GITHUB_ENV`?

### PR DESCRIPTION
**Pull request by mmlb**
I've added a GitHub Actions export only shell as I can't really see how a hook for it would be used or wanted.

Fixes #903

### CODE REVIEW

File: internal/cmd/shell.go [hunk 1]
```diff
@@ -44,20 +44,20 @@ func DetectShell(target string) Shell {
...
 		return JSON
-	case "elvish":
-		return Elvish
+	case "tcsh":
+		return Tcsh
+	case "vim":
+		return Vim
+	case "zsh":
+		return Zsh
 	}
```
Comment (mmlb) [inline internal/cmd/shell.go:63]: I hope this is alright, sorting sortable things is something I do out of habit. It does tend to be nice when looking things up, but can very much drop this commit if you don't wan the noise.

**Comment by mmlb** [comment]
Obviously no tests yet. It seems to me that the current test setup is geared heavily for interactive use of direnv via the shell hook setup. This doesn't really make sense for the gha format unless I should have a gha -> bash converter and let the common tests run? If not, then I'd like to know which tests I should implement? I would think maybe just comparing exported envs as redirected into a file?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -42,3 +42,18 @@ jobs:
         GO111MODULE: on
       run: make test-stdlib test-bash
 
+    - name: GitHub Actions Env Test Setup
```
Comment (mmlb) [inline .github/workflows/go.yml:45]: I was going to create a separate workflow file for this test but decided to just append the tests here instead. I can split it to different files if that makes more sense.

Should I "port" this test to do something when running locally on a dev machine? It feels kind of odd to only test some functionality in CI vs locally first.

**Comment by mmlb** [comment]
Hey @zimbatm gentle ping. Hopefully not causing noise for you but I'm not sure if you've seen this PR.

**Comment by BeyondEvil** [comment]
I've tested this in a self-hosted Ubuntu runner, and it works as expected 👍

**Comment by zimbatm** [comment]
Looks good to me. @mmlb do you mind checking why the CI is failing? Other than that, I'm happy with the changes.

**Comment by mmlb** [comment]
Hey @zimbatm I'm not sure why CI failed, I've added a debug commit to see if something pops up can you please release it for testing?

**Comment by mmlb** [comment]
Test failure might be due to $SRANDOM only existing in bash >5.1!

**Comment by mmlb** [comment]
ok @zimbatm I've punted on windows like the `go tests` and everything is passing again, PTAL.

File: .github/workflows/nix.yml [hunk 3]
```diff
@@ -1,8 +1,7 @@
 name: Nix
 on:
   push:
-    branches: [ master ]
-  pull_request: {}
```
[the remark this answers is not in the document]
Comment (mmlb) [reply .github/workflows/nix.yml:5]: Woops yes indeed. There's a similar change on go.yml, I presume that was is ok to keep?

[the remark this answers is not in the document]
Comment (zimbatm) [reply .github/workflows/nix.yml:5]: Both should use:
```
on:                                                                           
  push:                                                                       
    branches: [ master ]                                                      
  pull_request: {}
```
Once that's restore the PR is ready to merge :)

[the remark this answers is not in the document]
Comment (mmlb) [reply .github/workflows/nix.yml]: Done!

I'm curious why you want to limit the branches to just master though? Running the tests for all branches makes for an easier time for first time contributors while GH gates the CI builds. Though I guess those contributors could just PR off of master in their fork, but that doesn't seem to be the norm for most people.

[the remark this answers is not in the document]
Comment (zimbatm) [reply .github/workflows/nix.yml]: I kinda settled on this as a middle ground. As a maintainer, I want to have PRs be tested, and adding push on top adds more noise as both will be running then.

**Comment by nicknovitski** [comment]
This doesn't seem to work well with modifications to `PATH`, since github actions handles that differently.

I am trying an approach like this:

```yaml
- name: Load PATH changes
  run: |
        direnv_path_value="$(direnv exec . sh -c 'echo $PATH')"
        IFS=: # loop over PATH entries, which are separated by ":"
        for direnv_path_entry in $(echo "$direnv_path_value"); do
          if ! echo "$PATH" | grep "$direnv_path_entry" &> /dev/null; then
            echo "GITHUB_PATH does not include $direnv_path_entry, adding now"
            echo "$direnv_path_entry" >> "$GITHUB_PATH"
          fi
        done
- name: Load other environment changes
  run: direnv export gha >> "$GITHUB_ENV"
```

If I'm right, I could write a failing test to prove it, and maaaaybe add a `gha-path` export format??  I'm not sure.

**Comment by zimbatm** [comment]
if I understand what you mean properly, the step could be simplified to:
```yaml
- name: Load PATH changes
  run: direnv exec . sh -c 'echo $PATH' > "$GITHUB_PATH"
- name: Load other environment changes
  run: direnv export gha >> "$GITHUB_ENV"
```

**Comment by zimbatm** [comment]
Related issue: https://github.com/actions/toolkit/issues/655

**Pull request merged**